sui-execution/latest/sui-adapter/src/gas_charger.rs

```diff
@@ -244,6 +247,22 @@ pub mod checked {
             self.gas_status.charge_storage_read(total_size)
         }
 
+        pub fn charge_regulated_transfers(
+            &mut self,
+            num_deny_lists_checked: u64,
+            num_regulated_coin_owners: u64,
+        ) -> Result<(), ExecutionError> {
+            let cost_per_deny_list = CONFIG_DYNAMIC_OBJECT_FIELD_SIZE;
+            let deny_list_cost = cost_per_deny_list * (num_deny_lists_checked as usize);
+            // times two for the global pause and per-address settings
+            // we add the cost for the global pause per-address to minimize overcharging for
+            // non-regulated coins
+            let cost_per_owner = CONFIG_SETTING_DYNAMIC_FIELD_SIZE * 2;
+            let owner_cost = cost_per_owner * (num_regulated_coin_owners as usize);
+            self.gas_status
+                .charge_storage_read(deny_list_cost + owner_cost)
```

Comment:
@tzakian @dariorussi thoughts here?
I would really prefer to charge on the hashing we are doing but am unsure how to insert that cost

Reply:
you mean how to quantify that value? 
let's talk about it